src/main/java/de/tum/in/www1/artemis/service/notifications/GeneralInstantNotificationService.java

```diff
@@ -81,11 +79,8 @@ public void sendNotification(Notification notification, User user, Object notifi
      * @param users               who should be contacted
      * @param notificationSubject that is used to provide further information (e.g. exercise, attachment, post, etc.)
      */
-    @Async
```

Comment:
I think it's important that we keep async here to avoid blocking in case of many notifications being sent

Reply:
This method only filters the recipients, the actual notifications are sent using other async methods (`sendNotification`)